axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/OpportunitySaleOrderServiceImpl.java

```diff
@@ -20,36 +20,54 @@
 import com.axelor.apps.base.AxelorException;
 import com.axelor.apps.base.db.Company;
 import com.axelor.apps.base.db.Currency;
+import com.axelor.apps.base.db.repo.PartnerRepository;
 import com.axelor.apps.base.db.repo.PriceListRepository;
 import com.axelor.apps.base.db.repo.TraceBackRepository;
 import com.axelor.apps.base.exceptions.BaseExceptionMessage;
 import com.axelor.apps.base.service.PartnerPriceListService;
 import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.crm.db.LostReason;
 import com.axelor.apps.crm.db.Opportunity;
+import com.axelor.apps.crm.db.OpportunityStatus;
+import com.axelor.apps.crm.db.repo.OpportunityRepository;
+import com.axelor.apps.crm.db.repo.OpportunityStatusRepository;
+import com.axelor.apps.crm.service.OpportunityServiceImpl;
+import com.axelor.apps.crm.service.app.AppCrmService;
 import com.axelor.apps.sale.db.SaleOrder;
 import com.axelor.apps.sale.db.repo.SaleOrderRepository;
 import com.axelor.apps.sale.exception.SaleExceptionMessage;
 import com.axelor.i18n.I18n;
 import com.axelor.inject.Beans;
 import com.google.inject.Inject;
 import com.google.inject.persist.Transactional;
+import java.util.List;
 
-public class OpportunitySaleOrderServiceImpl implements OpportunitySaleOrderService {
+public class OpportunitySaleOrderServiceImpl extends OpportunityServiceImpl
```

Comment:
Please instead create a distinct service to extend OpportunityServiceImpl, called `OpportunityServiceSaleImpl"